### PR TITLE
Refactoring DelegatedAttribute

### PR DESCRIPTION
**Pull request by awead**
Day One of ActiveFedora Conclave. This refactors DelegatedAttribute into three different subclasses for better code organization.

### CODE REVIEW

**Comment by awead** [comment]
@hectorcorrea actually did most of this work, I just rebased it with some minor refactoring and adding documentation.

File: lib/active_fedora/delegated_attribute.rb [hunk 1]
```diff
@@ -2,105 +2,12 @@ module ActiveFedora
   # Represents the mapping between a model attribute and a field in a datastream
```
Comment (cam156) [inline lib/active_fedora/delegated_attribute.rb]: We need to update this comment.  This no longer represents a field in a stream, but a field in any object

**Comment by cjcolvar** [comment]
I'm confused as to the special treatment of ActiveTripleAttribute versus the rest of the attributes subclassing from DelegatedAttribute.  Is it just convenience to use :predicate instead of :delegate_to or is that a requirement of how AT works?  It is a bit confusing from a naming/code reading perspective for ActiveTripleAttributes to be added to delegated_attributes yet not be DelegatedAttributes.

**Comment by jcoyne** [comment]
@cjcolvar they are delegated attributes, they are just delegated to an ActiveTriples resource rather than a Datastream.

**Comment by cjcolvar** [comment]
How about renaming DelegatedAttribute (and associated methods) to something like DelegatedFileAttribute to clarify things?

**Comment by jcoyne** [comment]
@cjcolvar  DelegatedAttribute is the superclass of all the attributes.  DelegatedStreamAttribute is the superclass of OM + RDFDatastream attributes.

File: lib/active_fedora/attributes.rb [hunk 2]
```diff
@@ -215,31 +218,46 @@ def warn_duplicate_predicates new_name, new_properties
...
+      # @param [Symbol] field the field to find or create
+      # @param [Class] klass the class to use to delegate the attribute (e.g. 
+      #                ActiveTripleAttribute, OmAttribute, or RdfDatastreamAttribute)
+      # @param [Hash] args 
+      # @option args [String] :delegate_target the path to the delegate
+      # @option args [Class] :klass the class to create
+      # @option args [true,false] :multiple (false) true for multi-value fields
+      # @option args [Array<Symbol>] :at path to a deep node 
+      # @return [DelegatedAttribute] the found or created attribute
+      def find_or_create_defined_attribute(field, klass, args)
```
Comment (cjcolvar) [inline lib/active_fedora/attributes.rb]: It appears that this method is only being used for create and not find anymore.  Should the find aspect be removed?

**Comment by cjcolvar** [comment]
:clap: Looks good.  There are a few more "datastream" references that should probably be changed to "file" that could be addressed in this PR.

File: lib/active_fedora/attributes.rb (hunk 2 above)
Comment (jcoyne) [inline lib/active_fedora/attributes.rb]: It doesn't create one if one is found. Same as it did before.

**Pull request merged**